provider/core/src/response.rs

```diff
@@ -108,6 +115,19 @@ impl Cart {
     }
 }
 
+/// An empty cart sentinel value for the DataPayload. Better for memory
+/// layout than Option<Cart> in the DataPayload; see #4438.
+///
+/// Get one with the [`empty_cart()`] function.
+static EMPTY_CART: OnceCell<Cart> = OnceCell::new();
+
+/// Gets a cart that can serve as the empty cart in DataPayload.
+fn empty_cart() -> Cart {
```

Comment:
Yeah I don't think we can make this `const`. While we can very carefully construct const boxes (they're just a reference with a drop), there's no way for `Rc` because it has (private) structure on the heap.

Reply:
Fixed via no longer using `OnceCell`